src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint_configuration.py

```diff
@@ -0,0 +1,198 @@
+"""Entrypoint configuration for the k8s master/orchestrator pod."""
+
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import json
+from typing import Dict, List, Set
+
+from tfx.proto.orchestration.pipeline_pb2 import Pipeline as Pb2Pipeline
+
+from zenml.entrypoints.step_entrypoint_configuration import (
+    INPUT_ARTIFACT_SOURCES_OPTION,
+    MATERIALIZER_SOURCES_OPTION,
+    STEP_SOURCE_OPTION,
+)
+from zenml.integrations.kubernetes.orchestrators.kubernetes_step_entrypoint_configuration import (
+    RUN_NAME_OPTION,
+    KubernetesStepEntrypointConfiguration,
+)
+from zenml.steps.base_step import BaseStep
+
+PIPELINE_NAME_OPTION = "pipeline_name"
+IMAGE_NAME_OPTION = "image_name"
+NAMESPACE_OPTION = "kubernetes_namespace"
+PIPELINE_CONFIG_OPTION = "pipeline_config"
+
+STEP_SPECIFIC_STEP_ENTRYPOINT_OPTIONS = [
+    STEP_SOURCE_OPTION,
+    INPUT_ARTIFACT_SOURCES_OPTION,
+    MATERIALIZER_SOURCES_OPTION,
+]  # options from StepEntrypointConfiguration that change from step to step.
+
+
+def get_fixed_step_args(
```

Comment:
I think the naming of this function is a little confusing, as a boolean flag can force it to do the exact opposite of it's name. As this is only used once below (I think) and we need both the fixed and non-fixed arguments, do you think it's a good idea to rename (and rewrite) it to something like `def split_step_args(step_args) -> Tuple[List[str], List[str]]` that splits it into both fixed and non-fixed args?

Reply:
You're right, that is indeed confusing. I combined the two methods into one `split_step_args` as you suggested :)